pkg/controller/kafka/cluster/setup.go

```diff
@@ -117,7 +118,7 @@ func postObserve(_ context.Context, cr *svcapitypes.Cluster, obj *svcsdk.Describ
 }
 
 func preCreate(_ context.Context, cr *svcapitypes.Cluster, obj *svcsdk.CreateClusterInput) error {
-	obj.ClusterName = awsclients.String(meta.GetExternalName(cr))
+	obj.ClusterName = awsclients.String(cr.GetName())
```

Comment:
In other PRs we say: we can never use cr.Name, the source of truth is always the annotation. 
https://github.com/crossplane-contrib/provider-aws/pull/992#discussion_r764408679

Reply:
@chlunde any updates ?

Reply:
This change is correct. The `external-name` should be the uniq id of the resource, which is always the `ARN` in AWS which is provided by AWS.  (check the `postCreate()`: `meta.SetExternalName(cr, awsclients.StringValue(obj.ClusterArn))`)

However, the ClusterName is the user defined name. Which is currently not offered in parameters. I think it's reasonable to use the name of CR